File: libraries/logging/stream-logger/src/main/java/io/getstream/logging/TaggedLoggerImpl.java
package io.getstream.logging;

import static io.getstream.logging.StreamLogger.DEBUG;
import static io.getstream.logging.StreamLogger.ERROR;
import static io.getstream.logging.StreamLogger.INFO;
import static io.getstream.logging.StreamLogger.VERBOSE;
import static io.getstream.logging.StreamLogger.WARN;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

class TaggedLoggerImpl implements TaggedLogger {

  private final String tag;
  private final StreamLogger delegate;

  public TaggedLoggerImpl(String tag, StreamLogger logger) {
    this.tag = tag;
    this.delegate = logger;
  }

  @Override
  public void e(@NotNull Throwable throwable, @NotNull String message, @Nullable Object... args) {
    delegate.log(ERROR, tag, throwable, message, args);
  }

  @Override
  public void e(@NotNull String message, @Nullable Object... args) {
    delegate.log(ERROR, tag, null, message, args);
  }

  @Override
  public void w(@NotNull String message, @Nullable Object... args) {
    delegate.log(WARN, tag, null, message, args);
  }

  @Override
  public void i(@NotNull String message, @Nullable Object... args) {
    delegate.log(INFO, tag, null, message, args);
  }

  @Override
  public void d(@NotNull String message, @Nullable Object... args) {
    delegate.log(DEBUG, tag, null, message, args);
  }

  @Override
  public void v(@NotNull String message, @Nullable Object... args) {
    delegate.log(VERBOSE, tag, null, message, args);
  }
}

File: libraries/logging/stream-logger/src/main/java/io/getstream/logging/TaggedLoggerImpl.kt
/*
 * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
 *
 * Licensed under the Stream License;
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.getstream.logging

internal class TaggedLoggerImpl(
    private val tag: String,
    private val delegate: StreamLogger,
) : TaggedLogger {

    override fun e(throwable: Throwable, message: () -> String) {
        delegate.log(StreamLogger.ERROR, tag, message, throwable)
    }

    override fun e(message: () -> String) {
        delegate.log(StreamLogger.ERROR, tag, message)
    }

    override fun w(message: () -> String) {
        delegate.log(StreamLogger.WARN, tag, message)
    }

    override fun i(message: () -> String) {
        delegate.log(StreamLogger.INFO, tag, message)
    }

    override fun d(message: () -> String) {
        delegate.log(StreamLogger.DEBUG, tag, message)
    }

    override fun v(message: () -> String) {
        delegate.log(StreamLogger.VERBOSE, tag, message)
    }
}
